src/wallet/wallet.cpp

```diff
@@ -784,6 +784,27 @@ bool CWallet::IsSpentKey(const uint256& hash, unsigned int n) const
     return false;
 }
 
+void CWallet::SetUsedAddressState(WalletBatch& batch, const CTxOut& cout, bool used)
+{
+    AssertLockHeld(cs_wallet);
+    CTxDestination dst;
+    if(ExtractDestination(cout.scriptPubKey, dst)) {
+        if(IsMine(dst)) {
+            if (used && !GetDestData(dst, "address_used", nullptr)) {
+                AddDestData(batch, dst, "address_used", "p"); // p for "present", opposite of absent (null)
+            } else if (!used && GetDestData(dst, "address_used", nullptr)) {
+                EraseDestData(batch, dst, "address_used");
+            }
+        }
+    }
+}
+
+bool CWallet::IsUsedAddress(const CTxDestination& dst) const
+{
+    LOCK(cs_wallet);
+    return IsMine(dst) && GetDestData(dst, "address_used", nullptr);
```

Comment:
Is it me or `IsMine` is irrelevant?

Either way call `GetDestData` since it's faster and most likely to fail.

Reply:
Yes `IsMine` would be irrelevant in this case, have now removed to just return the output of `GetDestData` as suggested, Thanks!